test/controllers/unit_of_measures_controller_test.rb

```diff
@@ -72,8 +74,31 @@ class UnitOfMeasuresControllerTest < ActionDispatch::IntegrationTest
 end
 
 test "code must be unique" do
-  duplicate_unit_of_measure = @unit_of_measure.dup
-  @unit_of_measure.save
+  duplicate_unit_of_measure = @unit_of_measure_mt.dup
+  @unit_of_measure_mt.save
   assert_not duplicate_unit_of_measure.valid?
 end
+
+test "to_ref method in model" do
```

Comment:
@dawitd this is very good. My only comment is to move tests related to conversion to a separate file as they are not directly part of the controller.